lib/msgraph/errors.go

```diff
@@ -40,11 +41,21 @@ type graphErrorResponse struct {
 }
 
 // GraphError defines the structure of errors returned from MS Graph API.
+// https://learn.microsoft.com/en-us/graph/errors#json-representation
 type GraphError struct {
-	Code       string       `json:"code,omitempty"`
-	Message    string       `json:"message,omitempty"`
-	InnerError *GraphError  `json:"innerError,omitempty"`
-	Details    []GraphError `json:"details,omitempty"`
+	// Code is the code for the error, e.g. "UnknownError", "BadRequest".
+	Code string `json:"code,omitempty"`
+	// Message is a developer ready message about the error that occurred. This shouldn't be displayed
+	// to the user directly.
```

Comment:
Curious - shouldn't or its not ideal? What happens if this slips through user error message?

Reply:
I copied this from Graph docs. https://learn.microsoft.com/en-us/graph/errors#json-representation

We already show `Message` in the UI. 😶

https://github.com/gravitational/teleport/blob/1970fb19dae8242257af3e7af389687f48693b12/lib/msgraph/errors.go#L61-L69

So if it slips there is not the end of the world, but it might not be that readable to the end user, see https://github.com/gravitational/teleport.e/pull/7143.